app/subscriber/src/store/hooks/app/useApp.ts

```diff
@@ -82,7 +86,25 @@ export const useApp = (): [IAppState, IAppController] => {
     () => ({
       getUserInfo: async (refresh: boolean = false) => {
         if (userInfo.id !== 0 && !refresh) return userInfo;
-        const response = await dispatch('get-user-info', () => api.getUserInfo());
+        var location: IUserLocationModel;
+        try {
+          // Generate a unique key for this user and store in local storage.
+          var key = localStorage.getItem('device-key');
+          if (!key) {
+            key = crypto.randomUUID();
+            localStorage.setItem('device-key', key);
+          }
+          const locationResponse = await dispatch(
+            'get-location',
+            () => axios.get('https://geolocation-db.com/json/'),
```

Comment:
I don't think this is a great solution, but it's a neat one.

Reply:
Didn't even know that existed - 'tis neat!